src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -2909,6 +2927,46 @@ GIntBig QgsOgrLayer::GetApproxFeatureCount()
   return OGR_L_GetFeatureCount( hLayer, TRUE );
 }
 
+/**
+ * Returns the total (unfiltered) feature count according to OGRGeoPackageTableLayer::GetFeatureCount used by GDAL's ogrinfo.
+ */
+
+GIntBig QgsOgrLayer::getTotalFeatureCountfromMetaData() const
+{
+  // Don't quote column name (see https://trac.osgeo.org/gdal/ticket/5799#comment:9)
+  QByteArray layerName = OGR_L_GetName( hLayer );
+
+  QString driverName = GDALGetDriverShortName( GDALGetDatasetDriver( ds->hDS ) );
+  if ( driverName == QLatin1String( "GPKG" ) )
+  {
+    // QByteArray sql = "SELECT COUNT (*) FROM " + QgsOgrProviderUtils::quotedIdentifier( layerName, driverName );
```

Comment:
please remove that commented line of code and the other ones below

Reply:
done